v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandlerTest.java

```diff
@@ -0,0 +1,1028 @@
+    JSONObject newValuesJson = new JSONObject(newValuesString);
+    String colKey = "name";
+    handleCassandraAsciiType(colKey, newValuesJson);
+  }
+
+  @Test
+  public void testHandleValidStringVarint() {
+    String newValuesString = "{\"amount\":\"123456789123456789\"}";
+    JSONObject newValuesJson = new JSONObject(newValuesString);
+    String colKey = "amount";
+    BigInteger expected = new BigInteger("123456789123456789");
+    assertEquals(expected, handleCassandraVarintType(colKey, newValuesJson));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testHandleInvalidStringVarint() {
+    String newValuesString = "{\"amount\":\"abcxyz\"}";
+    JSONObject newValuesJson = new JSONObject(newValuesString);
+    String colKey = "amount";
+    handleCassandraVarintType(colKey, newValuesJson);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testHandleInvalidTypeVarint() {
+    String newValuesString = "{\"amount\":12345}";
+    JSONObject newValuesJson = new JSONObject(newValuesString);
+    String colKey = "amount";
+    handleCassandraVarintType(colKey, newValuesJson);
+  }
+
+  @Test
+  public void testHandleValidDuration() {
+    String newValuesString = "{\"duration\":\"P1DT1H\"}";
+    JSONObject newValuesJson = new JSONObject(newValuesString);
+    String colKey = "duration";
+    Duration expected = Duration.parse("P1DT1H");
+    assertEquals(expected, handleCassandraDurationType(colKey, newValuesJson));
+  }
+
+  @Test
+  public void testHandleNullDuration() {
+    String newValuesString = "{\"duration\":null}";
+    JSONObject newValuesJson = new JSONObject(newValuesString);
+    String colKey = "duration";
+    assertNull(handleCassandraDurationType(colKey, newValuesJson));
+  }
+
+  @Test
+  public void testHandleMissingColumnKey() {
+    String newValuesString = "{\"otherColumn\":\"P1DT1H\"}";
+    JSONObject newValuesJson = new JSONObject(newValuesString);
+    String colKey = "duration";
+    assertNull(handleCassandraDurationType(colKey, newValuesJson));
+  }
+
+  @Test
+  public void testHandleValidIPv4Address() throws UnknownHostException {
+    String newValuesString = "{\"ipAddress\":\"192.168.0.1\"}";
+    JSONObject newValuesJson = new JSONObject(newValuesString);
+    String colKey = "ipAddress";
+    InetAddress expected = InetAddress.getByName("192.168.0.1");
+    assertEquals(expected, handleCassandraInetAddressType(colKey, newValuesJson));
+  }
+
+  @Test
+  public void testHandleValidIPv6Address() throws UnknownHostException {
+    String newValuesString = "{\"ipAddress\":\"2001:0db8:85a3:0000:0000:8a2e:0370:7334\"}";
+    JSONObject newValuesJson = new JSONObject(newValuesString);
+    String colKey = "ipAddress";
+    InetAddress expected = InetAddress.getByName("2001:0db8:85a3:0000:0000:8a2e:0370:7334");
+    assertEquals(expected, handleCassandraInetAddressType(colKey, newValuesJson));
+  }
+
+  @Test(expected = UnknownHostException.class)
```

Comment:
in case you go with [com.google.common.net.InetAddresses](https://guava.dev/releases/20.0/api/docs/com/google/common/net/InetAddresses.html#forString-java.lang.String-), this will change to [IllegalArgumentException](http://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html?is-external=true)

Also Consider adding a case for `www.google.com`. In case we go with the  [com.google.common.net.InetAddresses](https://guava.dev/releases/20.0/api/docs/com/google/common/net/InetAddresses.html#forString-java.lang.String-), this will also throw `IllegalArgumentException` since the `google.common` package avoids DNS lookup

@shreyakhajanchi this would mean, that we support only ip addresses for inet columns and not host names 
I would treat this as a fair tradeoff given that Cassandra only supports  Inet that `Represents an IP address, either IPv4 (4 bytes long) or IPv6 (16 bytes long) `
with but we need to document that

Reply:
Please consider adding a test to check that "www.google.com" is treated as an invalid IP address to ensure that the underlying library we use does not attempt any DNS queries.